File: mobile/android/RoleplayAI/app/src/main/java/com/roleplayai/app/ai/MockEngine.kt
package com.roleplayai.app.ai

import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow

class MockEngine : AiEngine {
    override val name: String = "Mock (Offline)"

    override fun generateStream(prompt: String, settings: GenerationSettings): Flow<String> = flow {
        val text = "I am running locally. Wire llama.cpp or ONNX to replace this engine."
        for (ch in text) {
            emit(ch.toString())
            delay(8)
        }
    }
}

File: mobile/ios/RoleplayAI/App/AI/MockEngine.swift
import Foundation

final class MockEngine: AIEngine {
    let name: String = "Mock (Offline)"

    func generateStream(prompt: String, settings: GenerationSettings) -> AsyncStream<String> {
        AsyncStream { continuation in
            let text = "I am running locally. Wire llama.cpp, ORT GenAI, or CoreML to replace this engine."
            Task {
                for ch in text {
                    continuation.yield(String(ch))
                    try? await Task.sleep(nanoseconds: 8_000_000)
                }
                continuation.finish()
            }
        }
    }
}

